l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -188,7 +188,7 @@ contract L1ERC20Bridge is IL1ERC20Bridge, ReentrancyGuard {
         if (_l1Token == ETH_TOKEN_ADDRESS) {
             revert ETHDepositNotSupported();
         }
-        uint256 amount = _depositFundsToAssetRouter(msg.sender, IERC20(_l1Token), _amount);
+        uint256 amount = _approveFundsToAssetRouter(msg.sender, IERC20(_l1Token), _amount);
```

Comment:
I am not sure what it achieves. The L1NTV will still try to grab the funds from the user in the 

```
                uint256 expectedDepositAmount = _depositFunds(_originalCaller, IERC20(nativeToken), _depositAmount); // note if _originalCaller is this contract, this will return 0. This does not happen.
```

part

Reply:
We call the transferFundsToNTV on the L1AR first. This checks if the L1ERC20Bridge has allowance towards it. 

I flipped the order though in the L1AR, it should check the L1ERC20 bridge allowance first